IdentityCore/src/oauth2/aad_base/MSIDAADWebviewFactory.m

```diff
@@ -123,6 +123,17 @@ - (MSIDWebviewResponse *)oAuthResponseWithURL:(NSURL *)url
 #if AD_BROKER
     MSIDCBAWebAADAuthResponse *cbaResponse = [[MSIDCBAWebAADAuthResponse alloc] initWithURL:url context:context error:nil];
     if (cbaResponse) return cbaResponse;
+    
+    if ([url.absoluteString containsString:@"sso_nonce="])
+    {
+        NSString *ssoNonce = [[url msidQueryParameters] valueForKey:@"sso_nonce"];
+        if (![NSString msidIsStringNilOrBlank:ssoNonce])
+        {
+            NSDictionary *userInfo = @{@"sso_nonce" : ssoNonce};
+            *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorAuthorizationFailed, @"Nonce in JWT headers is likely expired, received SSO nonce redirect response.", nil, nil, nil, context.correlationId, userInfo, NO);
```

Comment:
check for if(error) or this will crash if nil error is passed.

Reply:
Thanks for catching it, added check.